shell.nix

```diff
@@ -83,38 +83,32 @@ in pkgs.mkShell {
         echo "The bcc egg $BCC_EGG does not exist. Maybe the python or bcc version is different?"
       fi
 
-      # autogen
-      alias a="sh autogen.sh"
-
-      # configure
-      alias c="./configure --with-boost-libdir=\$NIX_BOOST_LIB_DIR"
-      alias c_no-wallet="./configure --with-boost-libdir=\$NIX_BOOST_LIB_DIR --disable-wallet"
-      alias c_fast="./configure --with-boost-libdir=\$NIX_BOOST_LIB_DIR --disable-wallet --disable-tests --disable-fuzz --disable-bench -disable-fuzz-binary"
-      alias c_fast_wallet="./configure --with-boost-libdir=\$NIX_BOOST_LIB_DIR --disable-tests --disable-bench"
-
-      # make
-      alias m="make -j${jobs}"
-
-      # configure + make combos
-      alias cm="c && m"
-      alias cm_fast="c_fast && m"
-
-      # autogen + configure + make combos
-      alias acm="a && c && m"
-      alias acm_nw="a && c_no-wallet && m"
-      alias acm_fast="a && c_fast && m"
-      alias acm_fast_wallet="a && c_fast_wallet && m"
-
-      # tests
-      alias ut="make check"
-      # functional tests
-      alias ft="python3 test/functional/test_runner.py"
-      # all tests
-      alias t="ut && ft"
-
+      # Creates a new shorthand with the name $1 aliasing to the command $2
+      # We don't use aliases as they might not work in non-interactive shell sessions
+      # such as a CI. 
+      shorthand() {
+        eval "$1() { bash -c '$2'; }"
+        echo "alias $1 = $2"
```

Comment:
Ah, was surprised that it was still outputting "alias $1 = $2". Feels a bit fake, one can no longer do:
```console
$ alias t
bash: alias: t: not found
```

Feels more honest to strip away the alias and output a heading before:
```
...
adding bcc egg to PYTHONPATH: /nix/store/b41np9nwysnh2s4jdsxd4jywhjm4myz4-bcc-0.30.0/lib/python3.11/site-packages/bcc-0.30.0-py3.11.egg

Shorthands:
a = sh autogen.sh
c = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR
c_no-wallet = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR --disable-wallet
...
```

Reply:
Good point. Changed it to:

```
...
Shorthands:
  a = sh autogen.sh
  c = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR
  c_no-wallet = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR --disable-wallet
  c_fast = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR --disable-wallet --disable-tests --disable-fuzz --disable-bench -disable-fuzz-binary
  c_fast_wallet = ./configure --with-boost-libdir=$NIX_BOOST_LIB_DIR --disable-tests --disable-bench
  m = make -j8
  cm = c & m
  cm_fast = c_fast && m
  acm = a && c && m
  acm_nw = a && c_no-wallet && m
  acm_fast = a && c_fast && m
  acm_fast_wallet = a && c_fast_wallet && m
  ut = make check
  ft = python3 test/functional/test_runner.py
  t = ut && ft
...
```